scim/data_groups.go

```diff
@@ -0,0 +1,34 @@
+package scim
+
+import (
+	"context"
+
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/iam"
+	"github.com/databricks/terraform-provider-databricks/common"
+)
+
+// DataSourceGroups searches from groups based on filter
+func DataSourceGroups() common.Resource {
+	type groupData struct {
+		DisplayNameFilter string   `json:"filter,omitempty"`
+		DisplayNames      []string `json:"display_names,omitempty" tf:"computed,slice_set"`
+	}
+
+	return common.WorkspaceData(func(ctx context.Context, data *groupData, w *databricks.WorkspaceClient) error {
+		groupSearch := iam.ListGroupsRequest{}
+		if data.DisplayNameFilter != "" {
+			groupSearch = iam.ListGroupsRequest{Filter: data.DisplayNameFilter, Count: 100}
```

Comment:
just do it as:

```
groupSearch.Filter = data.DisplayNameFilter
```

And `Count` defines maximum returned results, not page size

Reply:
Are you sure `Count` is not used for page size in this particular api? This is what I saw in the [api model](https://github.com/databricks/databricks-sdk-go/blob/37cb031019b9cbe7e7388e6a14bcc6e54bbc13b5/service/iam/model.go#L486)

Reply:
Ah, you're right - SCIM API uses different paging method :-(